lib/testutils/client.go

```diff
@@ -77,18 +63,88 @@ func CreateNewIPPool(c client.Client, poolSubnet string, ipip, natOut, ipam bool
 	if err != nil {
 		log.Printf("Error creating pool: %s\n", err)
 	}
+}
+
+func CreateClient(config api.CalicoAPIConfig) *client.Client {
+	c, err := client.New(config)
+
+	if err != nil {
+		panic(err)
+	}
 
+	return c
 }
 
-// CleanIPPools removes all IP pool configuration from the datastore.
 func CleanIPPools(c *client.Client) {
 	if pools, err := c.IPPools().List(api.IPPoolMetadata{}); err == nil {
 		for _, pool := range pools.Items {
 			if err := c.IPPools().Delete(pool.Metadata); err != nil {
 				panic(err)
 			}
 		}
-	} else {
+	}
+}
+
+func CleanDatastore(config api.CalicoAPIConfig) {
+	var err error
+
+	log.Println(fmt.Sprintf("Cleaning datastore: %v", config.Spec.DatastoreType))
+
+	switch config.Spec.DatastoreType {
+	case api.EtcdV2:
+		cfg := etcdclient.Config{
```

Comment:
@robbrockbank config.EtcdConfig is config for our backend which is transformed to config of etcd client inside of etcd backend. I need to duplicate that logic here, because there is no option to get backend specific config from api config or from configured client.